### PR TITLE
Clean up health calculator

### LINKED ISSUE #227: New Default HP formula is malformed
```sourcepawn
KvGetString(BossKV[Special[client]], "health_formula", formula, sizeof(formula), "(((760.8+n)*n-1)^1.0341)+2046");
```

should be

```sourcepawn
KvGetString(BossKV[Special[client]], "health_formula", formula, sizeof(formula), "(((760.8+n)*n)^1.0341)+2046");
```

otherwise this spits out (i was testing with Saxton)

* [FF2] Saxton Hale has a malformed boss health formula, using default!

### PR DESCRIPTION
**Pull request by WildCard65**
Should close #227 IDK haven't tested.
But it does look a lot neater and SHOULD handle all cases of operation that old system didn't(eg: (5+2)*(5+3) as well as constant health values);

### CODE REVIEW

**Comment by WildCard65** [comment]
Calling @shadow93 to test this.

**Comment by WildCard65** [comment]
Forgot to mention this change also prepares for #164

**Comment by WildCard65** [comment]
@50Wliu I noticed Jenkins is not erroring once even though travis is.

**Comment by 50Wliu** [comment]
I'll have to take a look at that I guess.

**Comment by 50Wliu** [comment]
Should be fixed.  I've set up a separate job for pull requests.

**Comment by WildCard65** [comment]
Well at least travis works as intended which told me the problems

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 1]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
 	return -1;
 }
 
-stock CalcBossHealthMax(client)
+stock Calculate(client, const String:key[], const String:defaultFormula[], malformDefault)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: ParseFormula please

Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: defaultValue sounds better to me than malformDefault.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 2]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
-	new bool:mustClose, bool:usePlayers, bool:canAdd, bool:valueReady;
-	new parentheses;
-	new Float:sum[128];
-	new _operator[128];
-
-	KvRewind(BossKV[Special[client]]);
-	KvGetString(BossKV[Special[client]], "health_formula", formula, sizeof(formula), "(((760.8+n)*n-1)^1.0341)+2046");
+	new String:value[1024], Float:sum[128], calcOperators:_operator[128];
+	new bool:closeBracket, bool:usePlayers, bool:useLives, bool:foundOperator, bracket;
+	new Handle:bKv = BossKV[Special[client]];
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Can we reference this directly without creating a new variable for it?  That's how it's done everywhere else in the code.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 3]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
-				parentheses++;
-				sum[parentheses]=0.0;
-				_operator[parentheses]=0;
+				//Were in a bracket.
+				if (bracket + 1 == sizeof(_operator))
+				{
+					decl String:bossName[32];
+					KvRewind(bKv);
+					KvGetString(bKv, "name", bossName, sizeof(bossName));
+					LogError("[FF2] %s has too many operators in operation! Returning malformDefault", bossName);
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: %s's health formula is too complex, using default!

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 4]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
 		strcopy(buffer, sizeof(buffer), formula[i]);
-		switch(buffer[0])
+		switch (buffer[0])
 		{
 			case '(':
 			{
-				parentheses++;
-				sum[parentheses]=0.0;
-				_operator[parentheses]=0;
+				//Were in a bracket.
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Seems a bit self-explanatory due to the `case '('` above.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 5]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
-			case '-':
-			{
-				_operator[parentheses]=2;
-				canAdd=true;
+				//Close bracket now!
+				closeBracket = true;
+				break;
 			}
-			case '*':
+			default:
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Now this could use some explaining.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 6]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
 			{
-				_operator[parentheses]=3;
-				canAdd=true;
+				StrCat(value, sizeof(value), buffer);
+				break;
 			}
-			case '/':
+		}
+		if (i == length && !closeBracket && !foundOperator) //Handle constant operation(eg: "health_formula" 5000
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: If you could find a non-hacky way to figure out if we have a constant value that would be great :P.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 7]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
+		{
+			foundOperator = true;
+		}
+		if (foundOperator || closeBracket)
+		{
+			if (foundOperator)
 			{
-				_operator[parentheses]=4;
-				canAdd=true;
+				foundOperator = false;
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Why

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 8]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
 			}
-			case '^':
+			if (closeBracket)
 			{
-				_operator[parentheses]=5;
-				canAdd=true;
+				closeBracket = false;
+				bracket--;
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Why

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 9]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
+						break;
+					}
+					case Operation_Division:
+					{
+						sum[bracket] /= sum[bracket+1];
+						break;
+					}
+					case Operation_Exponential:
+					{
+						sum[bracket] ^= sum[bracket+1];
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: This works? o.O

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 10]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
+						break;
+					}
+					default:
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Why

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 11]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
+					{
+						sum[bracket] = sum[bracket+1];
+						break;
+					}
+				}
+				_operator[bracket+1] = Operation_None;
 			}
-			default:
+			if (usePlayers)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: This doesn't seem to get set anywhere.

Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Never mind we're in a for loop.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 12]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
-		if(valueReady)
-		{
-			valueReady=false;
-			if(usePlayers)
+			if (useLives)
 			{
-				usePlayers=false;
-				switch(_operator[parentheses])
+				useLives = false;
+				//Add sum with players.
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: .

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 13]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
+					}
+				}
 			}
-		}
-
-		if(valueReady)
-		{
-			valueReady=false;
-			if(usePlayers)
+			if (useLives)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Why do we need this?  Max lives are constant.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 14]
```diff
@@ -6543,182 +6553,243 @@ stock GetBossIndex(client)
...
 					{
-						parentheses=1;
+						sum[bracket] = float(BossLivesMax[Special[client]]);
 						break;
 					}
 				}
 			}
-
-			if(value[0]!='\0' && canAdd)
+			if (value[0] != '\0')
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: What are we doing here

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 13 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I always felt it be easier for boss makers if ff2 handles doing math involving lives

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 15]
```diff
@@ -6543,182 +6553,240 @@ stock GetBossIndex(client)
 	return -1;
 }
 
-stock CalcBossHealthMax(client)
+stock ParseFormula(client, const String:key[], const String:defaultFormula[], malformDefault)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: malformDefault-->defaultValue

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 16]
```diff
@@ -6543,182 +6553,240 @@ stock GetBossIndex(client)
...
+			case '(':
 			{
-				valueReady=true;
-				mustClose=true;
+				if (bracket + 1 == sizeof(_operator))
+				{
+					decl String:bossName[32];
+					KvRewind(BossKV[Special[client]]);
+					KvGetString(BossKV[Special[client]], "name", bossName, sizeof(bossName));
+					LogError("[FF2] %s health formula is too complex, using default!", bossName);
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: %s's

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 17]
```diff
@@ -6543,182 +6553,240 @@ stock GetBossIndex(client)
...
 			{
-				_operator[parentheses]=5;
-				canAdd=true;
+				_operator[bracket] = Operation_Exponential;
+				foundOperator = true;
+				break;
 			}
 			default:
 			{
+				//Character doesn't match things.
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: ??

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 18]
```diff
@@ -6543,182 +6553,240 @@ stock GetBossIndex(client)
...
+						break;
+					}
+				}
+			}
+			//TODO: Add useLives back.
+			if (value[0] != '\0') //Value has something in it, by time we get here we already have all the value parsed(hopefully).
+			{
+				new Float:tValue = StringToFloat(value);
+				strcopy(value, sizeof(value), "");
+				value[0] = '\0';
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Seems redundant

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 19]
```diff
@@ -6543,182 +6553,248 @@ stock GetBossIndex(client)
...
+						sum[bracket] = float(playing);
+						break;
 					}
-					case 4:
+				}
+			}
+			//TODO: Add useLives back.
+			if (value[0] != '\0') //Value has something in it, by time we get here we already have all the value parsed(hopefully).
+			{
+				new Float:tValue = StringToFloat(value);
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Perhaps the string could simply be called stringValue and the float value.  Or we could think of an entirely different name for the string.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 20]
```diff
@@ -6543,182 +6553,248 @@ stock GetBossIndex(client)
...
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= tValue;
+						break;
+					}
+					case Operation_Division:
+					{
+						if (tValue <= 0.0)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Why not less than 0?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 21]
```diff
@@ -6543,182 +6553,248 @@ stock GetBossIndex(client)
...
 						{
-							parentheses=1;
+							bracket = 0;
+							sum[bracket] = -999999.0; //Let malformation handle this.
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: This is a totally reasonable value (not really, but still :wink:).

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 22]
```diff
@@ -6543,182 +6553,248 @@ stock GetBossIndex(client)
...
-			parentheses--;
-			sum[parentheses]=sum[parentheses+1];
+			_operator[bracket] = Operation_None;
 		}
 	}
-
-	new health=RoundFloat(sum[0]);
-	if(!health && value[0]!='\0')  //Check to see if we're dealing with a constant health value
+	new result=RoundFloat(sum[0]);
+	if(!result && value[0]!='\0')  //Check to see if we're dealing with a constant health value
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: :( Is this still needed?  Can we find a way to get rid of it?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 23]
```diff
@@ -6543,182 +6553,248 @@ stock GetBossIndex(client)
...
-		LogError("[FF2] %s has a malformed boss health formula, using default!", bossName);
-		health=RoundFloat(Pow(((460.0+playing)*playing), 1.075));
+		LogError("[FF2] %s has a malformed boss %s formula, using default!", bossName, key);
+		return defaultValue;
 	}
-
 	if(bMedieval)
 	{
-		health=RoundFloat(health/3.6);  //TODO: Make this configurable
+		return RoundFloat(result/3.6);	//TODO: Make this configurable
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: What's the spacing change here?

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 22 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I have no idea, soul reason it's still in.

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 23 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: working on that, some how managed to mess up my local repo

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 19 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Last time I changed value, broke compile(forgot I renamed value)

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 24]
```diff
@@ -6543,182 +6553,247 @@ stock GetBossIndex(client)
...
-
-	if(parentheses || health<=0)
+	if(bracket || result<=0)
 	{
 		decl String:bossName[32];
 		KvRewind(BossKV[Special[client]]);
 		KvGetString(BossKV[Special[client]], "name", bossName, sizeof(bossName));
-		LogError("[FF2] %s has a malformed boss health formula, using default!", bossName);
-		health=RoundFloat(Pow(((460.0+playing)*playing), 1.075));
+		LogError("[FF2] %s has a malformed boss %s formula, using default!", bossName, key);
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I think you can get rid of "boss" now

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 25]
```diff
@@ -2708,7 +2718,7 @@ public Action:StartBossTimer(Handle:timer)
 	{
 		if(Boss[boss] && IsValidEdict(Boss[boss]) && IsPlayerAlive(Boss[boss]))
 		{
-			BossHealthMax[boss]=CalcBossHealthMax(boss);
+			BossHealthMax[boss]=ParseFormula(boss, "health_formula", "(((760.8+n)*n-1)^1.0341)+2046", RoundFloat(Pow(((460.0+playing)*playing), 1.075)));
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Your default value here is different than the default formula...;P

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Well then it's your fault cause I copy+pasted what was originally in ff2

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 19 above)
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I'd still like this to be changed-I'm sure the compiler errors can be worked out.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 26]
```diff
@@ -6543,182 +6553,247 @@ stock GetBossIndex(client)
...
+						break;
 					}
-					case 5:
+					case Operation_Exponential:
 					{
-						sum[parentheses]=Pow(sum[parentheses], Float:playing);
+						sum[bracket] ^= sum[bracket+1];
+						break;
 					}
 					default:
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: What are we doing here?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 27]
```diff
@@ -6543,182 +6553,247 @@ stock GetBossIndex(client)
...
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= tValue;
+						break;
+					}
+					case Operation_Division:
+					{
+						if (StrEqual(value, "0") || StrContains(value, "0.0") == 0)
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Why not a StringToInt here?  Invalid values will return 0 anyway.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 28]
```diff
@@ -6543,182 +6553,247 @@ stock GetBossIndex(client)
...
 						{
-							parentheses=1;
+							bracket = 0;
+							sum[bracket] = -999999.0; //Let malformation handle this.
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: We shouldn't be relying on this to indicate that the health formula is invalid.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 29]
```diff
@@ -2708,7 +2718,7 @@ public Action:StartBossTimer(Handle:timer)
 	{
 		if(Boss[boss] && IsValidEdict(Boss[boss]) && IsPlayerAlive(Boss[boss]))
 		{
-			BossHealthMax[boss]=CalcBossHealthMax(boss);
+			BossHealthMax[boss]=ParseFormula(boss, "health_formula", "(((760.8+n)*n-1)^1.0341)+2046", RoundFloat(((760.8+playing)*(playing-1.0))^FLOAT_PI)+2046.0));
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: 1.0341!=FLOAT_PI :wink:

File: addons/sourcemod/scripting/freak_fortress_2.sp (hunk 28 above)
Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Explain?

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 30]
```diff
@@ -2708,7 +2718,7 @@ public Action:StartBossTimer(Handle:timer)
 	{
 		if(Boss[boss] && IsValidEdict(Boss[boss]) && IsPlayerAlive(Boss[boss]))
 		{
-			BossHealthMax[boss]=CalcBossHealthMax(boss);
+			BossHealthMax[boss]=ParseFormula(boss, "health_formula", "(((760.8+n)*n-1)^1.0341)+2046", RoundFloat(Pow((760.8+playing)*(playing-1.0),1.0341)+2046.0));
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Point still stands-1.0341 is not pi.

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 31]
```diff
@@ -6543,182 +6553,247 @@ stock GetBossIndex(client)
...
-		if(valueReady)
+		if (foundOperator || closeBracket)
 		{
-			valueReady=false;
-			if(usePlayers)
+			if (foundOperator)
 			{
-				usePlayers=false;
-				switch(_operator[parentheses])
+				foundOperator = false; //This here is just so I don't forget to tell the code, "We already operated on."
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: What?

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: You wanted a comment, I'm horrible at adding good comments(I'm commentless 99% of the time when I code)

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 32]
```diff
@@ -6543,182 +6553,247 @@ stock GetBossIndex(client)
...
-						sum[parentheses]*=StringToFloat(value);
-						strcopy(value, sizeof(value), "");
+						sum[bracket] *= value;
+						break;
 					}
-					case 4:
+					case Operation_Division:
 					{
-						if(!StringToFloat(value))
+						if (!StringToInt(strValue) && StrContains(strValue, ".") == -1) //I can't find a good way to check if it's 0.0 due to things like 0.01 or 0.001
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: We're operating with floats right?  So shouldn't 0.001 be ok?

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: yes, but I need to check if there's a true 0 and StringToInt would truncate off the decimal so things like 0.1 and 0.001 would = 0

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Unless I do a hacky check how many numbers are after the decimal and multiply StringToFloat(strValue) by that then round it and compare

Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: But StringToFloat which is what you had wouldn't.

Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Which is what was previously in the code*

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: yes but then we run into the issue of float equality is most of the time NEVER ACCURATE! Due to how computers store floats, there is always an issue with doing == of floats.

Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: 0 is 0 no matter how you look at it WildCard ;P.  We're not comparing floats here, we're comparing to see if it's equal to 0 or not.  I'm pretty sure comparing to 0 will be *pretty* accurate.

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: then that would be we always say(dividing by 0.5 is invalid)

File: addons/sourcemod/scripting/freak_fortress_2.sp [hunk 33]
```diff
@@ -6543,182 +6553,258 @@ stock GetBossIndex(client)
...
+						break;
+					}
+					case Operation_Multiplication:
+					{
+						sum[bracket] *= value;
+						break;
+					}
+					case Operation_Division:
+					{
+						int dotInd = StrContains(strValue, ".");
```
Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Now I'm even more confused.  Can you please explain in detail why comparing StringToFloat to 0 won't work (on Steam if possible)?

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: http://docs.oracle.com/cd/E19957-01/806-3568/ncg_goldberg.html

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Another thing, 5.0 CAN BECOME 4.99999999999999999999

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Final thing which should clear up: 1.0 - 1.0 = 0.0000000000000000000000000000000000000000000000000000000000000000000000000000000000000001

Comment (50Wliu) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: I guess I'll take this now and conduct my own tests on it then.

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Computers store floats as an int WITH PRECISION.(3.05 = 305 precise 2)

Comment (WildCard65) [inline addons/sourcemod/scripting/freak_fortress_2.sp]: Updated last comment

**Comment by shadow93** [comment]
I'll just leave this here:

hl2 2015-03-17 09-06-51-732

**Comment by 50Wliu** [comment]
Yeah I know it has problems

**Comment by WildCard65** [comment]
As I said, I don't have a way to test it.

**Pull request merged**